### PR TITLE
[GO] Accept empty body on REST command endpoint

### LINKED ISSUE #1457: [Go] Command endpoint empty body
## 📖 Description

Per the spec, the rest transport should accept an empty body with the POST. Treat this as a value kind empty (or nil) bypassing all of the body validation/parsing. If there is a body it should still be subject to all the validation/marshalling.

---

## 🎯 Goal / Outcome

POST command accepts no body

---

## ✅ Definition of Done (DoD)
- [ ] Feature is implemented and works as expected
- [ ] Edge cases are handled
- [ ] Tests are added or updated (if applicable)
- [ ] Documentation is updated (if needed)
- [ ] Code is reviewed and approved

---

## 🛠️ Implementation Notes (Optional)

---

## 🔗 Related Issues / Links

---
## 🚧 Risks / Open Questions (Optional)

---

### PR DESCRIPTION
**Pull request by ZacharyDuaibesRV**
## 📖 Description
Per ST 2138, a POST to the REST command endpoint may omit the body to mean "no value". An empty body is now treated as a nil payload (matching the gRPC path) and bypasses body validation, while a non-empty body is still fully validated and unmarshalled. The parse decision is based on the actual body bytes (io.ReadAll) instead of Content-Length, which the server reports as -1 for chunked requests (so the old `ContentLength > 0` guard could silently drop a real body).

## ✅ Definition of Done (DoD)
- [x] Feature is implemented and works as expected
- [x] Edge cases are handled (empty body, chunked/unknown length, body read error)
- [x] Tests are added or updated
- [x] Documentation is updated
- [x] Code is reviewed and approved

## 🛠️ Implementation Notes
- `rest_transport.go`: `handleCommandEndpoint` reads the body and branches on `len(body)` — empty → nil payload, non-empty → parse + validate.
- `json_utils.go`: split `parseValueJSON` out of `ReadRequestJSON` so the handler can detect an empty body before reusing the Content-Type + protojson validation.
- `rest_transport_test.go`: added EmptyBodyIsNilPayload, ChunkedBodyIsParsed, and BodyReadError tests.
- `REST.md`: documented the empty-body behavior.

## 🔗 Related Issues / Links
Closes #1457

### CODE REVIEW

File: sdks/go/pkg/transports/rest/json_utils.go [hunk 1]
```diff
@@ -224,22 +224,27 @@ func ReadAssetRequestJSON(r *http.Request) (*protos.ExternalObjectPayload, error
...
-
-	data, err := io.ReadAll(r.Body)
-	if err != nil || err == io.EOF {
-		return nil, catena.StatusResult{Code: catena.StatusCodeInvalidArgument, Error: fmt.Sprintf("failed to read request body: %v", err)}
+	mediaType, _, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		return nil, catena.StatusResult{Code: catena.StatusCodeInvalidArgument, Error: fmt.Sprintf("invalid content type: %s", contentType)}
+	}
+	if mediaType != "application/json" {
+		return nil, catena.StatusResult{Code: catena.StatusCodeInvalidArgument, Error: fmt.Sprintf("unsupported content type: %s, expected application/json", mediaType)}
```
[the remark this answers is not in the document]
Comment (abrown-RV) [reply sdks/go/pkg/transports/rest/json_utils.go:248]: I'm ok with this helper returning statusresult if that's more convenient since its just a helper that lives in a transport and we can ensure its used properly. If the helper may return different status codes, can use StatusResult. If its really only one could use a go error and the call site knows to always encode as the one kind of http code.

**Comment by abrown-RV** [comment]
resolve your conflicts first please

File: sdks/go/pkg/transports/rest_transport.go [hunk 2]
```diff
@@ -937,12 +937,24 @@ func (t *RestTransport) handleCommandEndpoint(w http.ResponseWriter, r *http.Req
...
+	r.Body.Close()
+	if err != nil {
+		logger.Error("failed to read command payload", "error", err)
+		val, res := catena.ReplyError[catena.Value](catena.StatusCodeInvalidArgument, "invalid command payload")
+		t.writeHTTPResult(w, res, val)
+		return
+	}
+	if len(body) > 0 {
+		reqValue, parseErr := parseValueJSON(r.Header.Get("Content-Type"), body)
+		if parseErr.Code != catena.StatusCodeOk {
```
[the remark this answers is not in the document]
Comment (ZacharyDuaibesRV) [reply sdks/go/pkg/transports/rest_transport.go]: fixed: the parse check is now if parseErr.IsError()

File: sdks/go/pkg/transports/rest/json_utils.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZacharyDuaibesRV) [reply sdks/go/pkg/transports/rest/json_utils.go:248]: kept StatusResult

File: sdks/go/pkg/transports/rest/rest_transport_test.go [hunk 3]
```diff
@@ -867,6 +867,63 @@ func TestTransport_ExecuteCommand(t *testing.T) {
...
+
+		req := httptest.NewRequest(http.MethodPost, "/st2138-api/v1/0/command/test", strings.NewReader(`{"int32_value": 42}`))
+		req.Header.Set("Content-Type", "application/json")
+		req.ContentLength = -1 // simulate chunked / unknown-length body
+		rec := httptest.NewRecorder()
+		transport.mux.ServeHTTP(rec, req)
+
+		assertStatus(t, rec, http.StatusOK)
+		if receivedPayload == nil {
+			t.Error("expected chunked body to be parsed into a non-nil payload")
```
Comment (abrown-RV) [inline sdks/go/pkg/transports/rest/rest_transport_test.go]: test the body value
```suggestion
		if v, ok := receivedPayload.(int32); !ok || v != 42 {
			t.Errorf("expected chunked body to be parsed into payload 42, got %v (%T)", receivedPayload, receivedPayload)
```

File: sdks/go/pkg/transports/rest/rest_transport_test.go [hunk 4]
```diff
@@ -867,6 +867,63 @@ func TestTransport_ExecuteCommand(t *testing.T) {
...
+		}
+		if receivedPayload != nil {
+			t.Errorf("expected nil payload for empty body, got %v", receivedPayload)
+		}
+	})
+
+	// A body sent without a declared length (chunked transfer encoding) reports
+	// ContentLength == -1. The payload must still be read and parsed rather than
+	// silently dropped, so the handler must not gate parsing on Content-Length.
+	t.Run("ChunkedBodyIsParsed", func(t *testing.T) {
```
Comment (abrown-RV) [inline sdks/go/pkg/transports/rest/rest_transport_test.go]: not the most honest of names, this is not actually doing a chunked body (that would require a real server instead of the test one). Should name to something about making sure it can handle the unknown content-length. The code correctly doesn't even touch the content length, which is correct. So this is a regression test.

**Review by abrown-RV — CHANGES_REQUESTED** [review]

**Review by abrown-RV — APPROVED** [review]
